packages/react-storage/src/components/StorageBrowser/context/actions/types.ts

```diff
@@ -0,0 +1,22 @@
+export interface FolderItem {
+  key: string;
+  type: 'FOLDER';
+}
+
+export interface FileItem {
+  key: string;
+  lastModified: Date;
+  size: number;
+  type: 'FILE';
+}
+
+export type LocationItem = FileItem | FolderItem;
+
+export type Permission = 'READ' | 'READWRITE' | 'WRITE';
+export type LocationType = 'OBJECT' | 'PREFIX' | 'BUCKET';
+
+export interface LocationData<T = Permission> {
+  scope: string;
```

Comment:
Will this be the full location path?

Reply:
Yes, not sure if I'm really aligned with `scope` as the name since it feels kinda access grant-y but it does seem to align well with the underlying service as a whole